src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Unix.cs

```diff
@@ -43,6 +43,16 @@ private FileVersionInfo(string fileName)
         /// <returns>true if the file is a managed assembly; otherwise, false.</returns>
         private bool TryLoadManagedAssemblyMetadata()
         {
+            // First make sure it's a file we can actually read from.  Only regular files are relevant,
+            // and attempting to open and read from a file such as a named pipe file could cause us to
+            // hang (waiting for someone else to open and write to the file).
+            Interop.Sys.FileStatus fileStatus;
+            if (Interop.Sys.Stat(_fileName, out fileStatus) != 0 ||
+                (fileStatus.Mode & Interop.Sys.FileTypes.S_IFMT) != Interop.Sys.FileTypes.S_IFREG)
```

Comment:
> It does the same thing 

It doesn't do the same thing.  File.Exists doesn't require that the file type be S_IFREG; it'll also return true for S_IFFIFO.